### PR TITLE
IGNITE-15512 Append additional tests into ScriptTestSuite

### PR DESCRIPTION
**Pull request by zstan**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java [hunk 1]
```diff
@@ -607,4 +693,11 @@ private void checkResultsHashed(List<List<?>> res) {
         T,
         R
     }
+
+    /** */
+    private enum SortType {
+        ROWSORT,
```
Comment (korlov42) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java:699]: let's provide description for this sort types

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java [hunk 2]
```diff
@@ -595,6 +673,14 @@ private void checkResultsHashed(List<List<?>> res) {
         }
     }
 
+    /** */
+    public static String byteArrayToHex(byte[] arr) {
```
Comment (korlov42) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java:677]: IgniteUtils.byteArray2HexString()

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java [hunk 3]
```diff
@@ -585,8 +637,34 @@ private void checkEquals(String msg, String expectedStr, Object actual) {
 
         /** */
         private void checkResultsHashed(List<List<?>> res) {
-            // TODO:
-            throw new UnsupportedOperationException("Hashed results compare not supported");
+            Objects.requireNonNull(res, "empty result set");
+
+            messageDigest.reset();
+
+            for (List<?> l : res) {
```
Comment (korlov42) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java:644]: it's better to name it `row` and `col`

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java [hunk 4]
```diff
@@ -76,11 +81,29 @@
     /** Script. */
     private Script script;
 
+    /** nl to bytes representation. */
+    private static final byte[] NL_BYTES = "\n".getBytes();
```
Comment (korlov42) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java:85]: U.nl()

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java [hunk 5]
```diff
@@ -76,11 +81,29 @@
...
     public SqlScriptRunner(Path test, QueryEngine engine, IgniteLogger log) {
         this.test = test;
         this.engine = engine;
         this.log = log;
+
+        try {
+            messageDigest = MessageDigest.getInstance("MD5");
+        }
+        catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
```
Comment (korlov42) [inline modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java:103]: stack trace will be available in re-threw exception

**Review by korlov42 — APPROVED** [review]
Overall looks good except few minors

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java (hunk 2 above)
Comment (zstan) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java]: it converts to upper case, ok i extend utility method, thanks.

File: modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java (hunk 4 above)
Comment (ivandasch) [reply modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/logical/SqlScriptRunner.java:86]:
```
byte[] test = {10};
```
image

**Pull request closed without merging**